sdk/identity/azure-identity/azure/identity/_credentials/authorization_code.py

```diff
@@ -73,9 +73,11 @@ def get_token(self, *scopes, **kwargs):
         return token
 
     def _redeem_refresh_token(self, scopes, **kwargs):
-        # type: (Iterable[str], **Any) -> Optional[AccessToken]
+        # type: (Sequence[str], **Any) -> Optional[AccessToken]
         for refresh_token in self._client.get_cached_refresh_tokens(scopes):
-            token = self._client.obtain_token_by_refresh_token(refresh_token, scopes, **kwargs)
+            if "secret" not in refresh_token:
+                continue
+            token = self._client.obtain_token_by_refresh_token(scopes, refresh_token["secret"], **kwargs)
```

Comment:
Looks like we introduced breaking changes?

Reply:
I don't see one here, can you be more specific?

Reply:
refresh_token -> refresh_token["secret"]. This is very scary. Looks like we change the semantics?

Reply:
`refresh_token` is from the MSAL cache. It's a dict whose "secret" item is the actual token, as a string. One motivation of this refactoring is decreasing the coupling between `AadClient` and `msal`. To that end, I have `obtain_token_by_refresh_token` take the refresh token as a string rather than MSAL's dict.